### PR TITLE
feat: new elasticsearch instrumentation

### PR DESCRIPTION
**Pull request by YanivD**

### CODE REVIEW

File: packages/instrumentation-elasticsearch/package.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        "test": "mocha"
+    },
+    "bugs": {
+        "url": "https://github.com/aspecto-io/opentelemetry-ext-js/issues"
+    },
+    "dependencies": {
+        "@opentelemetry/api": "^0.17.0",
+        "@opentelemetry/instrumentation": "^0.17.0",
+        "@opentelemetry/semantic-conventions": "^0.17.0",
+        "test-all-versions": "^5.0.1"
```
Comment (nirsky) [inline packages/instrumentation-elasticsearch/package.json:44]: should be dev dependency

File: packages/instrumentation-elasticsearch/src/elasticsearch.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    InstrumentationModuleDefinition,
+    InstrumentationNodeModuleDefinition,
+    InstrumentationNodeModuleFile,
+} from '@opentelemetry/instrumentation';
+import { VERSION } from './version';
+import { DatabaseAttribute } from '@opentelemetry/semantic-conventions';
+import { startSpan, onError, onResponse, defaultDbStatementSerializer, normalizeArguments } from './utils';
+import { ELASTICSEARCH_API_FILES } from './helpers';
+
+type Config = InstrumentationConfig & ElasticsearchInstrumentationConfig;
```
Comment (nirsky) [inline packages/instrumentation-elasticsearch/src/elasticsearch.ts:16]: i actually changes this and used ElasticsearchInstrumentationConfig directly instead

File: packages/instrumentation-elasticsearch/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Span } from '@opentelemetry/api';
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+
+export interface SerializerPayload {
```
Comment (nirsky) [inline packages/instrumentation-elasticsearch/src/types.ts:4]: unused, copied from mongoose

File: packages/instrumentation-elasticsearch/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    condition?: any;
+    options?: any;
+    updates?: any;
+    document?: any;
+    aggregatePipeline?: any;
+}
+
+export type DbStatementSerializer = (params?: object, options?: object) => string;
```
Comment (nirsky) [inline packages/instrumentation-elasticsearch/src/types.ts:12]: could be nice to pass the operation as well, like in mongoose and redis

File: packages/instrumentation-elasticsearch/LICENSE [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Copyright 2021 @ Aspecto
```
Comment (nirsky) [inline packages/instrumentation-elasticsearch/LICENSE:1]: use apache license

File: packages/instrumentation-elasticsearch/package.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        "type": "git",
+        "url": "https://github.com/aspecto-io/opentelemetry-ext-js.git"
+    },
+    "scripts": {
+        "build": "tsc",
+        "prepare": "yarn run build",
+        "watch": "tsc -w",
+        "version:update": "node ../../scripts/version-update.js",
+        "version": "yarn run version:update",
+        "test": "mocha"
```
Comment (nirsky) [inline packages/instrumentation-elasticsearch/package.json:35]: shouldn't be invoked with `tav`?

File: package.json [hunk 7]
```diff
@@ -3,6 +3,7 @@
     "private": true,
     "scripts": {
         "test": "lerna run test",
+        "test:ci": "yarn test",
```
Comment (blumamir) [inline package.json:6]: lerna run test:ci

**Review by blumamir — APPROVED** [review]

File: package.json [hunk 8]
```diff
@@ -2,7 +2,7 @@
     "name": "root",
     "private": true,
     "scripts": {
-        "test": "lerna run test",
+        "test": "lerna run test:ci",
```
Comment (blumamir) [inline package.json:5]: should be `lerna run test`
and add another one called "test:ci": "lerna run test:ci"

**Pull request merged**